Libs/MRML/Core/vtkMRMLTransformStorageNode.cxx

```diff
@@ -447,8 +447,16 @@ int vtkMRMLTransformStorageNode::ReadDataInternal(vtkMRMLNode *refNode)
   // check that the file exists
   if (vtksys::SystemTools::FileExists(fullName.c_str()) == false)
     {
-    vtkErrorMacro("ReadDataInternal: transform file '" << fullName.c_str() << "' not found.");
-    return 0;
+    if (this->GetWriteState() == CancelledNoData)
```

Comment:
Should we even try to read if write state was CancelledNoData? Maybe the current behavior is better (e.g., you can prepopulate a scene and add a valid data file later), but a comment to clarify the reasoning would be helpful. It can be described here or at the description of the states.

Reply:
I'm not sure I understand this fully. What do you mean by current behavior? Do you suggest that I move this check out of the if FileExists block?

Reply:
If state is "CancelledNoData" but a data file exists by the name that is stored in the storage node. We call ReadData. Possible behaviors:
A. read the file (state has preference over presence of file).
B. do not read the file (presence of file has preference over the state).

I think currently behavior B is implemented consistently, but we should still specify that this is the intended behavior and why. We should probably also always consistently clear the data set if it is read from a "CancelledNoData" storage node.